lib/facet.ts

```diff
@@ -29,6 +29,8 @@ export interface AttributeMap {
 	[key: string]: AttributeValue;
 }
 
+export type Keys<T> = T extends T ? keyof T : never;
```

Comment:
Should this be in `keys.ts` instead of `facet.ts`?

Reply:
Yes, that might make more sense.